test/ShallowWrapper-spec.jsx

```diff
@@ -565,6 +565,36 @@ describe('shallow', () => {
       expect(wrapper.find('a[value=true]')).to.have.length(0);
     });
 
+    it('should support attributes with numbers', () => {
+      const wrapper = shallow(
+        <div>
+          <span value-1={1} />
+          <a value-1={false} />
+        </div>,
+      );
+
+      expect(wrapper.find('span[value-1=1]')).to.have.length(1);
```

Comment:
`span[value-1=1]` is not a valid CSS selector. Attribute values must be either a valid selector (which cannot start with a number) or a string ([reference](https://www.w3.org/TR/css3-selectors/#attribute-substrings)).

`span[value-1=1]` should fail and `span[value-1="1"]` should pass.